controllers/errorController.js

```diff
@@ -0,0 +1,72 @@
+const AppError = require("./../utils/appError");
+
+const handleCastErrorDB = (err) => {
+  const message = `Invalid ${err.path}: ${err.value}.`;
+  return new AppError(message, 400);
+};
+
+const handleDuplicateFieldsDB = (err) => {
+  // console.log(err);
+  //const value = err.keyValue.match(/(["'])(\\?.)*?\1/)[0];
+  //console.log(value);
+
+  const message = `Duplicate ${Object.keys(err.keyValue).map(
+    (el) => el
+  )} value: ${
+    err.keyValue.title || err.keyValue.email
+  }. Please use another value!`;
+  return new AppError(message, 400);
+};
+const handleValidationErrorDB = (err) => {
+  const errors = Object.values(err.errors).map((el) => el.message);
+  // console.log(errors);
+  const message = `Invalid input data. ${errors.join(". ")}`;
+  return new AppError(message, 400);
+};
+
+const sendErrorDev = (err, res) => {
+  res.status(err.statusCode).json({
+    status: err.status,
+    error: err,
+    message: err.message,
+    stack: err.stack,
+  });
+};
+
+const sendErrorProd = (err, res) => {
```

Comment:
What's the purpose of this?

Reply:
the error handling is based on development and production environments, on dev, the error is more detailed and technical while production is different.

Reply:
I disagree with the logic behind this. Production error should be as detailed as dev except if it's not an error but rather just information that does not mean anything.

Sometimes app runs fine on test and may not run well on prod. If you decide not to log error on prod, then debugging will be hell for you.